diff --git a/sdk/cpprt/linux/libunwind/autogen.sh b/sdk/cpprt/linux/libunwind/autogen.sh
--- a/sdk/cpprt/linux/libunwind/autogen.sh
+++ b/sdk/cpprt/linux/libunwind/autogen.sh
@@ -11,6 +11,8 @@ CFLAGS="$CFLAGS -std=c99 -fno-builtin -DHAVE_SGX=1 -fPIC -DUNW_LOCAL_ONLY -fdebu
 # Remove duplicated compiler options and filter out `-nostdinc'
 CFLAGS=`echo $CFLAGS | tr ' ' '\n' | grep -v nostdinc | tr '\n' ' '`
 export CFLAGS
+CPPFLAGS=-isystem"${SYSROOT_INCDIR}"
+export CPPFLAGS
 
 #Insert following codes into configure after add "-mfunction-return=thunk-extern -mindirect-branch-register" option, Or the "checking whether the C compiler works..." check will fail
 #Insert following codes into configure after add "-mfunction-return=thunk-extern -mindirect-branch-register" option, Or the "checking whether we are cross compiling... " check will fail
@@ -38,7 +40,8 @@ else
   sed -i "${line_start},${line_end} s/^_ACEOF/#pragma GCC push_options\r\n#pragma GCC optimize (\"-fomit-frame-pointer\")\r\nvoid __x86_return_thunk(){__asm__(\"ret\\\n\\\t\");}\r\nvoid __x86_indirect_thunk_rax(){__asm__(\"jmp \*%rax\\\n\\\t\");}\r\n#pragma GCC pop_options\r\n_ACEOF/" ./configure
 fi
 
-test -n "$NOCONFIGURE" || "$srcdir/configure" --enable-shared=no \
+test -n "$NOCONFIGURE" || "$srcdir/configure" --host=${TARGET_SYS} --build=${BUILD_SYS} \
+                                              --enable-shared=no \
                                               --disable-block-signals \
                                               --enable-debug=no \
                                               --enable-debug-frame=no \

diff --git a/sdk/simulation/uae_service_sim/linux/Makefile b/sdk/simulation/uae_service_sim/linux/Makefile
--- a/sdk/simulation/uae_service_sim/linux/Makefile
+++ b/sdk/simulation/uae_service_sim/linux/Makefile
@@ -57,7 +57,7 @@ RDRAND_MAKEFILE := $(RDRAND_LIBDIR)/Makefile
 
 EXTERNAL_LIB += -L$(RDRAND_LIBDIR) -lrdrand
 EXTERNAL_LIB += -L$(RDRAND_LIBDIR) -lrt
-EXTERNAL_LIB += -ldl -lcrypto
+EXTERNAL_LIB += -L"$(SYSROOT_LIBDIR)" -ldl -lcrypto
 
 vpath %.cpp $(LINUX_PSW_DIR)/ae/common \
             $(LINUX_SDK_DIR)/simulation/urtssim \
@@ -111,7 +111,7 @@ $(RDRAND_MAKEFILE):
 ifeq ($(ARCH), x86)
 	@cd $(RDRAND_LIBDIR);./configure
 else
-	@cd $(RDRAND_LIBDIR);./configure CFLAGS=-fPIC
+	@cd $(RDRAND_LIBDIR);./configure CFLAGS=-fPIC --host=$(TARGET_SYS) --build=$(BUILD_SYS)
 endif
 
 libsgx_%_deploy.so: ../libsgx_%_deploy.c
