diff --git a/sdk/encrypt_enclave/Makefile b/sdk/encrypt_enclave/Makefile
index d388dc1d..44457eb5 100644
--- a/sdk/encrypt_enclave/Makefile
+++ b/sdk/encrypt_enclave/Makefile
@@ -41,7 +41,7 @@ INC_DIR	:= -I$(COMMON_DIR)/inc -I$(COMMON_DIR)/inc/internal -I.
 CXXFLAGS += $(INC_DIR) -Wno-attributes -g -mrdrnd -fpie
 LDFLAGS := -pie $(COMMON_LDFLAGS)
 
-LINK_FLAGS := -lcrypto -L$(BUILD_DIR) -lsgx_tservice
+LINK_FLAGS := -L"$(SYSROOT_LIBDIR)" -lcrypto -L$(BUILD_DIR) -lsgx_tservice
 CPP_FILES := encryptip.cpp
 OBJS := $(CPP_FILES:.cpp=.o)
 TOOL_NAME := sgx_encrypt
diff --git a/sdk/sign_tool/SignTool/Makefile b/sdk/sign_tool/SignTool/Makefile
index e0d24164..f727bf1d 100644
--- a/sdk/sign_tool/SignTool/Makefile
+++ b/sdk/sign_tool/SignTool/Makefile
@@ -36,7 +36,7 @@ CFLAGS   += -Werror
 
 CFLAGS += -fpie -DOPENSSL_API_COMPAT=10101
 CXXFLAGS += -fpie -DOPENSSL_API_COMPAT=10101
-LDFLAGS := -pie $(COMMON_LDFLAGS)
+LDFLAGS := -pie $(COMMON_LDFLAGS)  -L$(SYSROOT_LIBDIR)
 
 INC += $(ADDED_INC)
 INC +=  -I$(COMMON_DIR)/inc                         \
