diff --git a/psw/ae/aesm_service/Makefile b/psw/ae/aesm_service/Makefile
index 1128b45..2f883dd 100644
--- a/psw/ae/aesm_service/Makefile
+++ b/psw/ae/aesm_service/Makefile
@@ -86,7 +86,7 @@ INCLUDE += -I$(SGX_IPP_INC) \
            -I$(AESM_INC_DIR) \
            -I$(IPC_COMMON_PROTO_DIR)
 
-EDGER8R  := $(LINUX_SDK_DIR)/edger8r/linux/_build/Edger8r.native
+EDGER8R  := sgx_edger8r
 
 
 SQLITECFLAGS := -fpie -Wall -Werror -Wno-maybe-uninitialized $(call cc-disable-warning,unused-const-variable)
diff --git a/sdk/edger8r/linux/Makefile b/sdk/edger8r/linux/Makefile
index 4706cac..1a0c9f6 100644
--- a/sdk/edger8r/linux/Makefile
+++ b/sdk/edger8r/linux/Makefile
@@ -43,9 +43,9 @@ all: build | $(BUILD_DIR)
 build:
 # ocaml 4.02 - added support for PIC
 ifeq ($(shell test $(OCAML_VERSION) -lt 402 && echo 1), 1)
-	ocamlbuild -lflag -ccopt -lflag "-Wl,-z,now" -no-links -libs str,unix  Edger8r.native
+	ocamlbuild -lflag -ccopt -lflag "-Wl,-z,now" -no-links -libs str,unix -lflags -cc,$(ROOT_DIR)/../recipe-sysroot-native/usr/bin/x86_64-ese-linux/x86_64-ese-linux-gcc,-ccopt,--sysroot=$(ROOT_DIR)/../recipe-sysroot Edger8r.native 
 else
-	ocamlbuild -cflags -ccopt,-fpie -lflags -runtime-variant,_pic,-ccopt,-pie,-ccopt -lflag "-Wl,-z,now"  -no-links -libs str,unix Edger8r.native
+	ocamlbuild -cflags -ccopt,-fpie -lflags -runtime-variant,_pic,-ccopt,-pie,-ccopt -lflag "-Wl,-z,now"  -no-links -libs str,unix -lflags -cc,$(ROOT_DIR)/../recipe-sysroot-native/usr/bin/x86_64-ese-linux/x86_64-ese-linux-gcc,-ccopt,--sysroot=$(ROOT_DIR)/../recipe-sysroot Edger8r.native
 endif
 
 $(BUILD_DIR):
diff --git a/sdk/protected_fs/sgx_tprotected_fs/Makefile b/sdk/protected_fs/sgx_tprotected_fs/Makefile
index 86a5104..28467be 100644
--- a/sdk/protected_fs/sgx_tprotected_fs/Makefile
+++ b/sdk/protected_fs/sgx_tprotected_fs/Makefile
@@ -45,8 +45,7 @@ CXXFLAGS += $(ENCLAVE_CXXFLAGS) -U__STRICT_ANSI__ -Werror
 SRC := $(wildcard *.cpp)
 OBJ := $(sort $(SRC:.cpp=.o))
 
-EDGER8R_DIR = $(LINUX_SDK_DIR)/edger8r/linux
-EDGER8R = $(EDGER8R_DIR)/_build/Edger8r.native
+EDGER8R = sgx_edger8r
 
 LIBNAME := libsgx_tprotected_fs.a
 
@@ -57,12 +56,9 @@ all: $(LIBNAME) | $(BUILD_DIR)
 $(LIBNAME): $(OBJ)
 	$(AR) rcsD $@ $(OBJ)
 	
-sgx_tprotected_fs_t.h: $(EDGER8R)
+sgx_tprotected_fs_t.h:
 	$(EDGER8R) --header-only --trusted $(COMMON_DIR)/inc/sgx_tprotected_fs.edl --search-path $(COMMON_DIR)/inc
 
-$(EDGER8R):
-	$(MAKE) -C $(EDGER8R_DIR)
-
 $(OBJ): %.o :%.cpp sgx_tprotected_fs_t.h
 	$(CXX) $(CXXFLAGS) $(INCLUDE)  -c $< -o $@
 
@@ -74,7 +70,6 @@ clean:
 	@$(RM) $(OBJ)
 	@$(RM) $(LIBNAME) $(BUILD_DIR)/$(LIBNAME)
 	@$(RM) sgx_tprotected_fs_t.*
-	$(MAKE) -C $(EDGER8R_DIR) clean
 .PHONY: rebuild
 rebuild: 
 	$(MAKE) clean 
diff --git a/sdk/protected_fs/sgx_uprotected_fs/Makefile b/sdk/protected_fs/sgx_uprotected_fs/Makefile
index 5c12152..a4f53fc 100644
--- a/sdk/protected_fs/sgx_uprotected_fs/Makefile
+++ b/sdk/protected_fs/sgx_uprotected_fs/Makefile
@@ -41,8 +41,7 @@ CXXFLAGS += -fPIC -fno-rtti -Werror $(INCLUDE)
 SRC := $(wildcard *.cpp)
 OBJ := $(sort $(SRC:.cpp=.o))
 
-EDGER8R_DIR = $(LINUX_SDK_DIR)/edger8r/linux
-EDGER8R = $(EDGER8R_DIR)/_build/Edger8r.native
+EDGER8R = sgx_edger8r
 
 LIBNAME := libsgx_uprotected_fs.a
 
@@ -53,12 +52,9 @@ all: $(LIBNAME) | $(BUILD_DIR)
 $(LIBNAME): $(OBJ)
 	$(AR) rcsD $@ $(OBJ)
 	
-sgx_tprotected_fs_u.h: $(EDGER8R)
+sgx_tprotected_fs_u.h:
 	$(EDGER8R) --header-only --untrusted $(COMMON_DIR)/inc/sgx_tprotected_fs.edl --search-path $(COMMON_DIR)/inc
 
-$(EDGER8R):
-	$(MAKE) -C $(EDGER8R_DIR)
-
 $(OBJ): %.o :%.cpp sgx_tprotected_fs_u.h
 	$(CXX) $(CXXFLAGS) $(INCLUDE)  -c $< -o $@
 
@@ -70,7 +66,6 @@ clean:
 	@$(RM) $(OBJ)
 	@$(RM) $(LIBNAME) $(BUILD_DIR)/$(LIBNAME)
 	@$(RM) sgx_tprotected_fs_u.*
-	$(MAKE) -C $(EDGER8R_DIR) clean
 
 .PHONY: rebuild
 rebuild:
diff --git a/sdk/simulation/tae_service_sim/Makefile b/sdk/simulation/tae_service_sim/Makefile
index e464024..1c4fd8a 100644
--- a/sdk/simulation/tae_service_sim/Makefile
+++ b/sdk/simulation/tae_service_sim/Makefile
@@ -56,8 +56,7 @@ SE_LIB        := $(SE_LIB_DIR)/libselib.a
 EC_DH_LIB_DIR := $(LINUX_SDK_DIR)/ec_dh_lib
 EC_DH_LIB     := $(EC_DH_LIB_DIR)/libsgx_dh.a
 
-EDGER8R_DIR   := $(LINUX_SDK_DIR)/edger8r/linux
-EDGER8R       := $(EDGER8R_DIR)/_build/Edger8r.native
+EDGER8R       := sgx_edger8r
 
 SRC     := $(wildcard *.cpp)
 SRC     += $(TOP_DIR)/common/src/dh.cpp
@@ -80,15 +79,12 @@ $(TSERVICE_SIM_LIB): $(TAE_SERVICE_SIM_LIB) $(EC_DH_LIB) $(TSEAL_LIB) $(SE_LIB)
 	$(AR) rsD $@ $(BUILD_DIR)/.tservice_sim/*.o
 	@$(RM) -rf  $(BUILD_DIR)/.tservice_sim
 
-sgx_tae_service_t.h: $(EDLFILE) $(EDGER8R)
+sgx_tae_service_t.h: $(EDLFILE)
 	$(EDGER8R) --trusted --header-only --search-path $(TOP_DIR)/common/inc $<
 
 $(OBJS): %.o: %.cpp sgx_tae_service_t.h
 	$(CXX) $(CXXFLAGS) -fno-exceptions -fno-rtti $(INCLUDE) $(DEFINES) -c $< -o $@
 
-$(EDGER8R):
-	$(MAKE) -C $(EDGER8R_DIR)
-
 $(EC_DH_LIB):
 	$(MAKE) -C $(EC_DH_LIB_DIR)
 
@@ -103,7 +99,6 @@ $(BUILD_DIR):
 
 .PHONY: clean
 clean:
-	$(MAKE) -C $(EDGER8R_DIR)   clean
 	$(MAKE) -C $(EC_DH_LIB_DIR) clean
 	$(MAKE) -C $(TSEAL_LIB_DIR) clean
 	$(MAKE) -C $(SE_LIB_DIR)    clean
diff --git a/sdk/tae_service/Makefile b/sdk/tae_service/Makefile
index 6cb83d1..48638d0 100644
--- a/sdk/tae_service/Makefile
+++ b/sdk/tae_service/Makefile
@@ -54,8 +54,7 @@ vpath %.cpp $(COMMON_DIR)/src:\
 
 CXXFLAGS += $(ENCLAVE_CXXFLAGS)
 
-EDGER8R_DIR    := $(LINUX_SDK_DIR)/edger8r/linux
-EDGER8R        := $(EDGER8R_DIR)/_build/Edger8r.native
+EDGER8R        := sgx_edger8r
 
 EC_DH_LIB_DIR := $(TOP_DIR)/sdk/ec_dh_lib
 EC_DH_LIB     := $(EC_DH_LIB_DIR)/libsgx_dh.a
@@ -70,12 +69,9 @@ LIBNAME := libtae_service.a
 .PHONY: all
 all: $(LIBNAME)
 
-$(AENAME)_t.h: $(EDLFILE) $(EDGER8R)
+$(AENAME)_t.h: $(EDLFILE)
 	$(EDGER8R) --trusted --header-only --search-path $(TOP_DIR)/common/inc $<
 
-$(EDGER8R):
-	$(MAKE) -C $(EDGER8R_DIR)
-
 $(LIBNAME): $(OBJ)
 	$(MAKE) -C $(EC_DH_LIB_DIR)
 	$(CP) $(EC_DH_LIB) $@
@@ -92,7 +88,6 @@ clean:
 	@$(RM) *_t.* *_u.*
 	@$(RM) $(LIBNAME)
 	$(MAKE) -C $(EC_DH_LIB_DIR) clean
-	$(MAKE) -C $(EDGER8R_DIR) clean
 
 .PHONY: rebuild
 rebuild:
diff --git a/sdk/tkey_exchange/Makefile b/sdk/tkey_exchange/Makefile
index eac8ad3..6c4fd2f 100644
--- a/sdk/tkey_exchange/Makefile
+++ b/sdk/tkey_exchange/Makefile
@@ -58,8 +58,7 @@ SRC += ecp.cpp
 
 OBJ := $(sort $(SRC:.cpp=.o))
 
-EDGER8R_DIR = $(LINUX_SDK_DIR)/edger8r/linux
-EDGER8R = $(EDGER8R_DIR)/_build/Edger8r.native
+EDGER8R = sgx_edger8r
 LIBNAME := libsgx_tkey_exchange.a
 
 .PHONY: all
@@ -73,12 +72,9 @@ $(LIBNAME): $(OBJ)
 
 tkey_exchange.o: $(COMMON_DIR)/inc/sgx_tkey_exchange_t.h
 
-$(COMMON_DIR)/inc/sgx_tkey_exchange_t.h: $(COMMON_DIR)/inc/sgx_tkey_exchange.edl $(EDGER8R)
+$(COMMON_DIR)/inc/sgx_tkey_exchange_t.h: $(COMMON_DIR)/inc/sgx_tkey_exchange.edl
 	$(EDGER8R) --search-path $(TOP_DIR)/common/inc sgx_tkey_exchange.edl
 
-$(EDGER8R):
-	$(MAKE) -C $(EDGER8R_DIR)
-
 $(OBJ): %.o :%.cpp
 	$(CXX) $(CXXFLAGS) $(INCLUDE)  -c $< -o $@
 
@@ -92,7 +88,6 @@ clean:
 	@$(RM) $(COMMON_DIR)/inc/sgx_tkey_exchange_t.h $(COMMON_DIR)/inc/sgx_tkey_exchange_u.h
 	@$(RM) $(COMMON_DIR)/src/sgx_tkey_exchange_t.c $(COMMON_DIR)/src/sgx_tkey_exchange_u.c
 	@$(RM) sgx_tkey_exchange_t.* sgx_tkey_exchange_u.*
-	$(MAKE) -C $(EDGER8R_DIR) clean
 
 .PHONY: rebuild
 rebuild: 
