diff --git a/psw/ae/common/src/UnixCommunicationSocket.cpp b/psw/ae/common/src/UnixCommunicationSocket.cpp
index 649c3a4c..42e46496 100644
--- a/psw/ae/common/src/UnixCommunicationSocket.cpp
+++ b/psw/ae/common/src/UnixCommunicationSocket.cpp
@@ -215,7 +215,7 @@ bool UnixCommunicationSocket::init()
         memset(&serv_addr, 0, sizeof(struct sockaddr_un));
         serv_addr.sun_family = AF_UNIX;
         memset(serv_addr.sun_path, 0, sizeof(serv_addr.sun_path));
-        strncpy(serv_addr.sun_path, mSocketBase, sizeof(serv_addr.sun_path));
+        strncpy(serv_addr.sun_path, mSocketBase, sizeof(serv_addr.sun_path) - 1);
 
         if( connect(mSocket, (struct sockaddr *)&serv_addr, sizeof(serv_addr)) != 0)
         {

